### PR TITLE
[WIP] wicked: refactor basic suite

### PR DESCRIPTION
**Pull request by jlausuch**
- Related ticket: https://progress.opensuse.org/issues/42641
- Proof run: http://fromm.arch.suse.de/tests/2902

NOTE: there is a test case failing due to a product bug already reported: https://bugzilla.suse.com/show_bug.cgi?id=1114875

to be merged after GMC

### CODE REVIEW

**Comment by asmorodskyi** [comment]
LGTM , but probably we need prove runs

File: tests/wicked/basic/t01_basic.pm [hunk 1]
```diff
@@ -63,4 +63,8 @@ sub run {
     validate_script_output("wicked show dev $iface",     sub { m/\[dhcp\]/g; });
 }
 
+sub test_flags {
+    return {always_rollback => 1, wicked_need_sync => 1};
```
Comment (cfconrad) [inline tests/wicked/basic/t01_basic.pm]: We do not need `wicked_need_sync` as long as we do not have a SUT and REF job.
Same on the other files.

```suggestion
    return {always_rollback => 1};
```

File: tests/wicked/basic/t02_static_addresses_legacy.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+use base 'wickedbase';
+use strict;
+use testapi;
+use network_utils 'iface';
+
+sub run {
+    my ($self) = @_;
+    my $iface = iface();
+    $self->{iface} = $iface;
```
Comment (cfconrad) [inline tests/wicked/basic/t02_static_addresses_legacy.pm]: Why? Same on the other files.

File: tests/wicked/basic/t06_static_routes_legacy.pm [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+sub run {
+    my ($self) = @_;
+    my $iface = iface();
+    $self->{iface} = $iface;
+    record_info('Info', 'Set up static routes from legacy ifcfg files');
+    $self->get_from_data('wicked/static_address/ifcfg-eth0',   "/etc/sysconfig/network/ifcfg-$iface");
+    $self->get_from_data('wicked/static_address/ifroute-eth0', "/etc/sysconfig/network/ifroute-$iface");
+    assert_script_run("ifup $iface");
+    $self->assert_wicked_state(ping_ip => '10.0.2.2', iface => $iface);
+    validate_script_output("ip -4 route show", sub { m/default via 10.0.2.2/ });
```
Comment (cfconrad) [inline tests/wicked/basic/t06_static_routes_legacy.pm]: A dot outside of `[]` is any character. `\.` would match the dot.

**Comment by jlausuch** [comment]
after loosing track of the original forked repo, I duplicated this PR https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6151

**Pull request closed without merging**